modules/3d/test/test_pointcloud.cpp

```diff
@@ -0,0 +1,87 @@
+#include <opencv2/core.hpp>
+#include <iostream>
+#include <vector>
+#include <string>
+
+#include "test_precomp.hpp"
+#include "opencv2/ts.hpp"
+
+namespace opencv_test { namespace {
+
+TEST(obj_test, point_cloud_load_tests)
+{
+    std::vector<cv::Point3f> points_1 = {
+        {-5.93915f, -0.13257f, 2.55837f},
+        {-5.93915f, 1.86743f, 2.55837f},
+        {-5.93915f, -0.13257f, -1.16339f},
+        {-5.93915f, 1.86743f, -1.16339f},
+        {0.399941f, -0.13257f, 2.55837f},
+        {0.399941f, 1.86743f, 2.55837f},
+        {0.399941f, -0.13257f, -1.16339f},
+        {0.399941f, 1.86743f, -1.16339f}};
+    std::vector<cv::Point3f> normals_1 = {
+        {-1.0000f, 0.0000f, 0.0000f},
+        {0.0000f, 0.0000f, -1.0000f},
+        {1.0000f, 0.0000f, 0.0000f},
+        {0.0000f, 0.0000f, 1.0000f},
+        {0.0000f, -1.0000f, 0.0000f},
+        {0.0000f, 1.0000f, 0.0000f}};
+
+    std::vector<cv::Point3f> points;
+    std::vector<cv::Point3f> normals;
+
+    std::string folder = string(cvtest::TS::ptr()->get_data_path()) + "pointcloudio/";
+    std::string original_path = folder + "orig.obj";
+
+    cv::loadPointCloud(original_path, points, normals);
+
+    EXPECT_EQ(1,(points_1==points));
+    EXPECT_EQ(1,(normals_1==normals));
+}
+
+TEST(obj_test, point_cloud_save_tests)
+{
+    std::vector<cv::Point3f> points;
+    std::vector<cv::Point3f> normals;
+
+    std::string folder = std::string(cvtest::TS::ptr()->get_data_path()) + "pointcloudio/";
+    std::string original_path = folder + "orig.obj";
+    std::string new_path = folder + "new.obj";
+
+    cv::loadPointCloud(original_path, points, normals);
+
+    cv::savePointCloud(new_path, points, normals);
+
+    std::vector<cv::Point3f> points_1;
+    std::vector<cv::Point3f> normals_1;
+
+    cv::loadPointCloud(new_path, points_1, normals_1);
+
+    EXPECT_EQ(1,(normals_1==normals));
+    EXPECT_EQ(1,(points_1==points));
+}
+
+TEST(ply_test, point_cloud_load_tests)
+{
+    std::vector<cv::Point3f> points;
+    std::vector<cv::Point3f> normals;
+
+    std::string folder = std::string(cvtest::TS::ptr()->get_data_path()) + "pointcloudio/";
+    std::string original_path = folder + "orig.ply";
+    std::string new_path = folder + "new.ply";
+
+    cv::loadPointCloud(original_path, points, normals);
+
+    cv::savePointCloud(new_path, points, normals);
```

Comment:
It's bad idea to save things to OPENCV_TEST_DATA_PATH location:
- It confuses git. Developer can commit intermediate state by mistake
- In case of intermediate function failure the test always reads old file and indicates wrong PASS.

Please take a look on `cv::FileStorage` tests in core. They create temporary files in tmp location with different names and remove them after the test execution.